src/ophyd_epics_devices/zebra.py

```diff
@@ -0,0 +1,302 @@
+from enum import Enum
+from functools import partialmethod
+from typing import List
+
+import numpy as np
+import numpy.typing as npt
+from ophyd.v2.core import StandardReadable
+from ophyd.v2.epics import EpicsSignalR, EpicsSignalRW
+
+
+class Active(Enum):
+    no = "No"
+    yes = "Yes"
+
+
+class GatePulseSelect(Enum):
+    time = "Time"
+    position = "Position"
+    external = "External"
+
+
+class ArmSelect(Enum):
+    soft = "Soft"
+    external = "External"
+
+
+class Direction(Enum):
+    positive = "Positive"
+    negative = "Negative"
+
+
+class EncoderType(Enum):
+    enc1 = "Enc1"
+    enc2 = "Enc2"
+    enc3 = "Enc3"
+    enc4 = "Enc4"
+    enc1_4av = "Enc1-4Av"
+
+
+class TimeUnits(Enum):
+    ms = "ms"
+    s = "s"
+    s10 = "10s"
+
+
+class Capture(StandardReadable):
+    def __init__(self, prefix: str, name="") -> None:
+        self.enc1 = EpicsSignalRW(Active, "B0")
+        self.enc2 = EpicsSignalRW(Active, "B1")
+        self.enc3 = EpicsSignalRW(Active, "B2")
+        self.enc4 = EpicsSignalRW(Active, "B3")
+        self.sys1 = EpicsSignalRW(Active, "B4")
+        self.sys2 = EpicsSignalRW(Active, "B5")
+        self.div1 = EpicsSignalRW(Active, "B6")
+        self.div2 = EpicsSignalRW(Active, "B7")
+        self.div3 = EpicsSignalRW(Active, "B8")
+        self.div4 = EpicsSignalRW(Active, "B9")
+        super().__init__(prefix, name)
+
+
+class ArrayOuts(StandardReadable):
+    def __init__(self, prefix: str, name="") -> None:
+        def make_pv(suffix: str, common_prefix: str = "PC_"):
+            return EpicsSignalRW(npt.NDArray[np.float64], common_prefix + suffix)
+
+        self.enc1 = make_pv("ENC1")
+        self.enc2 = make_pv("ENC2")
+        self.enc3 = make_pv("ENC3")
+        self.enc4 = make_pv("ENC4")
+        self.sys1 = make_pv("SYS1")
+        self.sys2 = make_pv("SYS2")
+        self.div1 = make_pv("DIV1")
+        self.div2 = make_pv("DIV2")
+        self.div3 = make_pv("DIV3")
+        self.div4 = make_pv("DIV4")
+        super().__init__(prefix, name)
+
+
+class ZebraOutputPanel(StandardReadable):
+    def __init__(self, prefix: str, name="") -> None:
+        self.pulse_1_input = EpicsSignalRW(int, "PULSE1_INP")
+        self.out_ttl = {
+            channel: EpicsSignalRW(int, f"OUT{channel}_TTL") for channel in range(1, 5)
+        }
+        super().__init__(prefix, name)
+
+    @property
+    def out_ttl_list(self) -> List[EpicsSignalRW[int]]:
```

Comment:
Nit: I'm happy to drop the list now we have the dict, otherwise it'll just get confusing

Reply:
That's fine - I was wondering if this would be the case. Will remove it now.

Reply:
is the same true of GateControl.sources_list? We now have a self.sources dictionary in that which does the same thing.

Reply:
Yep, thanks, I missed that one